_scripts/evaluate_chains.py

```diff
@@ -0,0 +1,182 @@
+import json
+
+_PROVIDER_MAP = {
+    "openai": ChatOpenAI,
+    "anthropic": ChatAnthropic,
+}
+
+_MODEL_MAP = {
+    "openai": "gpt-3.5-turbo",
+    "anthropic": "claude-2",
+}
+
+def create_chain(
+    retriever: BaseRetriever,
+    model_provider: Union[Literal["openai"], Literal["anthropic"]],
+    chat_history: Optional[list] = None,
+    model: Optional[str] = None,
+    temperature: float = 0.0,
+) -> Runnable:
+    model_name = model or _MODEL_MAP[model_provider]
+    model = _PROVIDER_MAP[model_provider](model=model_name, temperature=temperature)
+    
+    _template = """Given the following conversation and a follow up question, rephrase the follow up question to be a standalone question.
+
+    Chat History:
+    {chat_history}
+    Follow Up Input: {question}
+    Standalone Question:"""
+    
+    
+    CONDENSE_QUESTION_PROMPT = PromptTemplate.from_template(_template)
+    
+    _template = """
+    You are an expert programmer and problem-solver, tasked to answer any question about Langchain. Using the provided context, answer the user's question to the best of your ability using the resources provided.
+    If you really don't know the answer, just say "Hmm, I'm not sure." Don't try to make up an answer.
+    Anything between the following markdown blocks is retrieved from a knowledge bank, not part of the conversation with the user. 
+    <context>
+        {context} 
+    <context/>"""
+
+    if chat_history:
+        _inputs = RunnableMap(
+                {
+                    "standalone_question": {
+                        "question": lambda x: x["question"],
+                        "chat_history": lambda x: x["chat_history"],
+                    } | CONDENSE_QUESTION_PROMPT | model | StrOutputParser(),
+                    "question": lambda x: x["question"],
+                    "chat_history": lambda x: x["chat_history"],
+                }
+            )
+        _context = {
+            "context": itemgetter("standalone_question") | retriever,
+            "question": lambda x: x["question"], 
+            "chat_history": lambda x: x["chat_history"],
+        }
+        prompt = ChatPromptTemplate.from_messages([
+            ("system", _template),
+            MessagesPlaceholder(variable_name="chat_history"),
+            ("human", "{question}"),
+        ])
+    else:
+        _inputs = RunnableMap(
+            {
+                "question": lambda x: x["question"],
+                "chat_history": lambda x: [],
+            }
+        )
+        _context = {
+            "context": itemgetter("question") | retriever,
+            "question": lambda x: x["question"], 
+            "chat_history": lambda x: [],
+        }
+        prompt = ChatPromptTemplate.from_messages([
+            ("system", _template),
+            ("human", "{question}"),
+        ])
+    
+    chain = (
+        _inputs
+        | _context
+        | prompt 
+        | ChatOpenAI(model="gpt-4", temperature=temperature)
```

Comment:
Should be swappable with Claude (based on what the user picks in the MODEL_MAP)?